src/main/java/backend/resource/MultiModel.java

```diff
@@ -207,15 +207,34 @@ private void preprocessUpdatedIssues(List<Model> newModels) {
                     // TODO O(n^2), optimise by preprocessing into a map or sorting
                     if (!existingModel.getIssueById(i).equals(model.getIssueById(i))) {
                         assert model.getIssueById(i).isPresent();
-                        // It's no longer currently read, but it retains its updated time.
-                        // No changes to preferences.
-                        model.getIssueById(i).get().setIsCurrentlyRead(false);
+                        /** If the issue was marked read after the last update, but update came in later
+                         * due to network latency, ensure the issue retains its read state
+                        **/
+                        if (updateReadAtState(model.getIssueById(i))) {
+                            model.getIssueById(i).get().setIsCurrentlyRead(false);
+                        }
                     }
                 }
             }
         }
     }
```

Comment:
This method has too much nesting. Nesting can be reduced by reversing `if` conditions.
Instead of 
```
if(condition){
   //do stuff
}
```
use,
```
if(!condition) return;
//do stuff
```